src/features/HomePage/components/Welcome.tsx

```diff
@@ -2,40 +2,51 @@ import styled, { css } from 'styled-components';
 import { useNavigate } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 
+import { selectChatsExist } from '@/features/Chat/chatSlice';
+import { selectUserRole } from '@/features/Authentication/userSlice';
+import { useAppSelector } from '@/store';
+
 import { palette } from '@/components/variables';
 import Text from '@/components/Text';
 import { TextButton } from '@/components/Buttons';
 
-import type { UserRole } from '@/features/Authentication/userSlice';
+type Role = 'admin' | 'freshMentee' | 'mentor' | 'mentee';
 
 type Props = {
   isMobile?: boolean;
-  role: UserRole;
 };
 
-const Welcome = ({ isMobile = false, role }: Props) => {
+const Welcome = ({ isMobile = false }: Props) => {
   const { t } = useTranslation('home');
+  const userRole = useAppSelector(selectUserRole);
+  const userHasChats = useAppSelector(selectChatsExist);
```

Comment:
We could simplify the ui-component slightly.
Instead of two separate selectors, we can just combine these:

```
export const selectUserRole = createSelector(
  selectUserState,
  selectChatsExist,
  ({ userRole }, hasUserChats) =>
    userRole === 'mentee' && !hasUserChats ? 'freshMentee' : userRole,
);
```

then use it liek this:
```
  const role = useAppSelector(selectUserRole);
```

The signature will be 
<img width="594" alt="image" src="https://github.com/user-attachments/assets/2b54c7f0-4671-4e8b-8912-c6a3e4346694">

Notice that we dont even need to define the type Role that extens the type UserRole.

Reply:
Great refactoring idea! I'll implement this!